### PR TITLE
Implement workaround for NAG compiler bug

### LINKED ISSUE #49: Nagfor: problem to compile sort.f90
Unfortunately, the NAG compiler can not compile toml-f, as it does not like the procedure pointer assignment in the pure `sort_keys` subroutine in sort.F90. I think, it is a compiler bug, and I've reported it already to NAG. However, it would be nice to have a quick workaround to enable testing with NAG for DFTB+ with tblite included.

Possible workarounds:
* Leave away the `pure` attribute from `sort_keys`
* Leave away the pointer assignment and put directly the `quicksort()` into the two branches of the if.

Thanks!

### PR DESCRIPTION
**Pull request by aradi**
Fixes #49

### CODE REVIEW

File: src/tomlf/utils/sort.f90 [hunk 1]
```diff
@@ -68,13 +67,11 @@ pure subroutine sort_keys(list, idx, compare)
       allocate(indexarray(high), source=[(i, i=low, high)])
 
       if (present(compare)) then
-         compare_func => compare
+        call quicksort(sorted, indexarray, low, high, compare)
```
Comment (awvwgk) [inline src/tomlf/utils/sort.f90:70]:
```suggestion
         call quicksort(sorted, indexarray, low, high, compare)
```

File: src/tomlf/utils/sort.f90 [hunk 2]
```diff
@@ -68,13 +67,11 @@ pure subroutine sort_keys(list, idx, compare)
...
       else
-         compare_func => compare_keys_less
+        call quicksort(sorted, indexarray, low, high, compare_keys_less)
```
Comment (awvwgk) [inline src/tomlf/utils/sort.f90:72]:
```suggestion
         call quicksort(sorted, indexarray, low, high, compare_keys_less)
```

**Review by awvwgk — APPROVED** [review]
Thanks for sharing. Looks good to me.

**Comment by aradi** [comment]
Sorry, I've overlooked the differing indentation, thx.

**Comment by awvwgk** [comment]
Looks like it is time to drop some old GCC versions from the testing.

**Comment by aradi** [comment]
GCC 5 (!),  come on, it was already due long time ago! :wink:

**Pull request merged**